### DISCUSSION TITLE
Kafka Companion: Which is the best way to verify messages are commited?

### DISCUSSION THREAD

**Discussion by cj19** [smallrye/smallrye-reactive-messaging#2669] [Q&A]
I am aware of the approach of listing the partition offsets:

```
TopicPartition topicPartition = KafkaCompanion.tp("topic", 1);
long latestOffset = companion.offsets().get(topicPartition, OffsetSpec.latest()).offset();
```

but what is the difference between latest and earliest offset? what is the best way to make sure the message got commited after producing it into one topic?

**Comment by ozangunalp** [comment]
You can use `companion.consumerGroups().offsets(consumerGroupId, listOfPartitions)` to check for committed offsets.
